RedHatInsights/vulnerability-ui
# VULN-1590 VULN-1286 feat: Read-only mode

Show notification and banner when write request returns 503 error. Banner is persistent between pages within Vulnerability.
![notif](https://user-images.githubusercontent.com/8426204/130635876-7aa356d2-7c70-40e2-8657-f7569f5039f8.png)
![banner](https://user-images.githubusercontent.com/8426204/130635866-8ab14295-bce9-47fb-b5fb-271011a8975c.png)

List of requests possibly returning 503 of Vulnerability:
- edit status
- edit pair status
- edit business risk
- exclude/include system

## Technical notes
I added a new property to `window` global variable - `window.showReadOnlyBanner`, because I didn't know how else to pass this function to `interceptors.js`. `errorInterceptor` in `interceptors.js` is responsible for showing the banner, but notification is shown by the particular components.

## How to test
Make sure your write request fails, for example edit `CveStatusModal` so it sets `statusId` to illegal value, like 1000, this will return 400 error so change occurences of 503 to 400 in 'interceptors.js' and 'BaseModal'.
